### PR TITLE
MONGOID-4947 include block and pattern params in any?

### PR DESCRIPTION
**Pull request by Neilshweky**
Apologies for all the whitespace changes in the diff.

### CODE REVIEW

**Comment by johnnyshields** [comment]
This is a regression introduced in 7.1. Please merge soon, thanks!

File: lib/mongoid/association/referenced/has_many/enumerable.rb [hunk 1]
```diff
@@ -219,11 +219,18 @@ def empty?
...
+            # here.
+            #
             # @example Doess the enumerable have any?
             #   enumerable.any?
             #
             # @return [ true, false ] If the enumerable has any elements.
-            def any?
+            def any?(pattern=nil, &block)
+              return super(pattern) if pattern
+              return super(&block) if block_given?
```
Comment (p-mongo) [inline lib/mongoid/association/referenced/has_many/enumerable.rb:232]: Both of these calls should forward both arguments to super, i.e. I believe they can be simplified to 

```
return super if pattern || block_given?
```

File: spec/mongoid/association/referenced/has_many/enumerable_spec.rb [hunk 2]
```diff
@@ -256,6 +256,111 @@
...
+        end
+
+        it "returns false when the predicate is false" do
+          expect(
+            enumerable.any? { |doc| false }
+          ).to be false
+        end
+      end
+
+      context "when an option is given" do
```
Comment (p-mongo) [inline spec/mongoid/association/referenced/has_many/enumerable_spec.rb:291]: s/option/argument/ and in the below tests

**Comment by p-mongo** [comment]
@johnnyshields Which Mongoid version is behaving correctly and which version is behaving incorrectly?

**Comment by johnnyshields** [comment]
@p-mongo version 7.0.x is fine. Also version 3.x is fine. I haven't tested 4/5/6

File: lib/mongoid/association/referenced/has_many/enumerable.rb (hunk 1 above)
Comment (Neilshweky) [reply lib/mongoid/association/referenced/has_many/enumerable.rb:232]: This doesn't seem to work, actually.

Comment (egiurleo) [reply lib/mongoid/association/referenced/has_many/enumerable.rb:232]: When you say it doesn't work, what behavior are you observing?

Comment (Neilshweky) [reply lib/mongoid/association/referenced/has_many/enumerable.rb:232]: The block is not being passed along in the super call

Comment (egiurleo) [reply lib/mongoid/association/referenced/has_many/enumerable.rb:232]: Thanks for explaining!

Comment (p-mongo) [reply lib/mongoid/association/referenced/has_many/enumerable.rb:232]: You still need to pass both arguments in both cases, you could try `super(pattern, &block)` if needed.

**Comment by Neilshweky** [comment]
@p-mongo for some reason this still causes failures... Also I don't really need to pass both arguments in both cases, because in Enumerble's implementation of any? if there is both a pattern and a block, the block is ignored.

**Pull request merged**